rust/hyperlane-base/src/settings/mod.rs

```diff
@@ -265,6 +278,7 @@ impl Settings {
             gelato: self.gelato.clone(),
             metrics: self.metrics.clone(),
             tracing: self.tracing.clone(),
+            defaultsigner: self.defaultsigner.clone(),
```

Comment:
This feels a bit weird to me given we (IIUC) put the default signer in the chains post deser.

Like I wonder if we should add `signer: Option<string>` to RelayerSettings and then just have the relayer modify the chains in-place

Reply:
I thought about putting it into the relayer but it would have been less pretty. This seemed like a good solution and also the signer is part of the chain configuration which is in the base config, not in the relayer config.